websocket-ballerina/tests/03_push_and_onText.bal

```diff
@@ -17,153 +17,40 @@
 import ballerina/runtime;
 import ballerina/test;
 import ballerina/io;
-import ballerina/http;
 
 string data = "";
-string expectedMsg = "{\"name\":\"Riyafa\", \"age\":23}";
 
-public type WebSocketPerson record {|
-   string name;
-   int age;
-|};
+listener Listener l2 = checkpanic new(21003);
 
-service UpgradeService /onTextString on new Listener(21003) {
-   remote isolated function onUpgrade(http:Caller caller, http:Request req) returns Service|WebSocketError {
+service /onTextString on l2 {
+   resource function onUpgrade .() returns Service|UpgradeError {
        return new WsService1();
    }
 }
 
 service class WsService1 {
   *Service;
-  remote isolated function onText(Caller caller, string data, boolean finalFrame) {
-      checkpanic caller->pushText(data);
+  remote isolated function onString(Caller caller, string data) {
+      checkpanic caller->writeString(data);
   }
 }
 
-service UpgradeService /onTextJSON on new Listener(21023) {
-   remote isolated function onUpgrade(http:Caller caller, http:Request req) returns Service|WebSocketError {
-       return new WsService2();
-   }
-}
-
-service class WsService2 {
-  *Service;
-  remote isolated function onText(Caller caller, json data) {
-      checkpanic caller->pushText(data);
-  }
-}
-
-service UpgradeService /onTextXML on new Listener(21024) {
-   remote isolated function onUpgrade(http:Caller caller, http:Request req) returns Service|WebSocketError {
-       return new WsService3();
-   }
-}
-
-service class WsService3 {
-  *Service;
-  remote isolated function onText(Caller caller, xml data) {
-      checkpanic caller->pushText(data);
-  }
-}
-
-service UpgradeService /onTextRecord on new Listener(21025) {
-    remote isolated function onUpgrade(http:Caller caller, http:Request req) returns Service|WebSocketError {
-       return new WsService4();
-   }
-}
-
-service class WsService4 {
-  *Service;
-  remote isolated function onText(Caller caller, WebSocketPerson data) {
-       var personData = data.cloneWithType(json);
-       if (personData is error) {
-           panic personData;
-       } else {
-           var returnVal = caller->pushText(personData);
-           if (returnVal is WebSocketError) {
-               panic <error>returnVal;
-           }
-       }
-   }
-}
-
-service UpgradeService /onTextByteArray on new Listener(21026) {
-    remote isolated function onUpgrade(http:Caller caller, http:Request req) returns Service|WebSocketError {
-       return new WsService5();
-    }
-}
-
-service class WsService5 {
-  *Service;
-  remote isolated function onText(Caller caller, byte[] data) {
-       var returnVal = caller->pushText(data);
-       if (returnVal is WebSocketError) {
-           panic <error>returnVal;
-       }
-   }
-}
-
-service object {} clientPushCallbackService = service object {
-    remote function onText(Client wsEp, string text) {
+service class clientPushCallbackService {
+    remote function onString(AsyncClient wsEp, string text) {
```

Comment:
Here what we need is the caller right not the AsyncClient.

Reply:
The caller is for the server-side. Here we pass in the created WebSocket client.

Reply:
Interesting. I thought we should expose the same client in the form of the caller.

Reply:
I think it should be the caller even when it is on the client side. Is there a specific reason to do otherwise ?

Reply:
We can rename this to Caller. But isn't it confusing for the users? They have created an AsyncClient and for sending messages they are using something called Caller?

Reply:
As per the discussion had, renaming this to Caller